src/worker/token.worker.ts

```diff
@@ -1,49 +1,66 @@
-import { MISSING_REFRESH_TOKEN_ERROR_MESSAGE } from './constants';
+import { MISSING_REFRESH_TOKEN_ERROR_MESSAGE } from '../constants';
+import { WorkerRefreshTokenMessage } from './worker.types';
 
-let refreshTokens = {};
+let refreshTokens: Record<string, string> = {};
 
-const cacheKey = (audience, scope) => `${audience}|${scope}`;
+const cacheKey = (audience: string, scope: string) => `${audience}|${scope}`;
 
-const getRefreshToken = (audience, scope) =>
+const getRefreshToken = (audience: string, scope: string) =>
   refreshTokens[cacheKey(audience, scope)];
 
-const setRefreshToken = (refreshToken, audience, scope) =>
-  (refreshTokens[cacheKey(audience, scope)] = refreshToken);
+const setRefreshToken = (
+  refreshToken: string,
+  audience: string,
+  scope: string
+) => (refreshTokens[cacheKey(audience, scope)] = refreshToken);
 
-const deleteRefreshToken = (audience, scope) =>
+const deleteRefreshToken = (audience: string, scope: string) =>
   delete refreshTokens[cacheKey(audience, scope)];
 
-const wait: any = time => new Promise(resolve => setTimeout(resolve, time));
+const wait = (time: number) =>
+  new Promise(resolve => setTimeout(resolve, time));
+
+const messageHandler = async (m: MessageEvent<WorkerRefreshTokenMessage>) => {
```

Comment:
Let me see if I can destructure it and still keep the type information.

Reply:
I've done this: 

```js
const messageHandler = async ({
  data: { timeout, auth, fetchUrl, fetchOptions },
  ports: [port]
}: MessageEvent<WorkerRefreshTokenMessage>) => {
  let json: {
    refresh_token?: string;
  };

  const { audience, scope } = auth || {};

  // ...
}
```

I'll keep the destructuring of `auth` out of the method signature; I find doing this in a method sig can get unreadable at times, and this will just make it worse.

Reply:
Yep 👍